py/picca/data.py

```diff
@@ -247,21 +248,33 @@ def mask(self,mask_obs,mask_RF):
         if not hasattr(self,'ll'):
             return
 
-        w = sp.ones(self.ll.size).astype(bool)
+        w = sp.ones(self.ll.size,dtype=bool)
         for l in mask_obs:
-            w = w & ( (self.ll<l[0]) | (self.ll>l[1]) )
+            w &= (self.ll<l[0]) | (self.ll>l[1])
         for l in mask_RF:
-            w = w & ( (self.ll-sp.log10(1.+self.zqso)<l[0]) | (self.ll-sp.log10(1.+self.zqso)>l[1]) )
+            w &= (self.ll-sp.log10(1.+self.zqso)<l[0]) | (self.ll-sp.log10(1.+self.zqso)>l[1])
 
-        self.ll = self.ll[w]
-        self.fl = self.fl[w]
-        self.iv = self.iv[w]
-        if self.mmef is not None:
-            self.mmef = self.mmef[w]
-        if self.diff is not None:
-             self.diff = self.diff[w]
-        if self.reso is not None:
-             self.reso = self.reso[w]
+        ps = ['iv','ll','fl','T_dla','T','mmef','diff','reso']
+        for p in ps:
+            if hasattr(self,p) and (getattr(self,p) is not None):
+                setattr(self,p,getattr(self,p)[w])
+
+        return
+
+    def add_optical_depth(self,tau,gamma,waveRF):
+        """Add mean optical depth
+        """
+        if not hasattr(self,'ll'):
+            return
+
+        if self.T is None:
+            self.T = sp.ones(self.ll.size)
+
+        w = 10.**self.ll/(1.+self.zqso)<=waveRF
+        z = 10.**self.ll/waveRF-1.
+        self.T[w] *= sp.exp(-tau*(1.+z[w])**gamma)
```

Comment:
ok, so T[w] is the expected flux at w given an overall tau and a gamma for the evolution assuming one has only one species (i.e. if one uses the function in the Lyb range without providing a Lyb optical depth this will not generate any).

The only thing I'd see myself whining about is tau being defined at z=0, I'd rather use z=3 or something (could also be a free parameter) as the anchorpoint as that's closer to the actual observed redshifts and both parameters are thus less correlated when trying to produce a given mean transmission at a given redshift. 
But the best choice might well be depending on what this function should be used for.

I'd also call the parameters A_tau (or tau0 or tau3) and eta_tau (as in Palanque 2015 or alternatively beta as in Becker 2013) or something to avoid confusion with other parameters that are commonly called gamma.

And I'd maybe switch T[w] -> Fbar[w].

Reply:
>ok, so T[w] is the expected flux at w given an overall tau and a gamma for the evolution assuming one has only one species (i.e. if one uses the function in the Lyb range without providing a Lyb optical depth this will not generate any).

Indeed, it adds one specie at a time, if the user adds more than one set of optical depth parameter, then all of them are added one at a time.

> The only thing I'd see myself whining about is tau being defined at z=0, I'd rather use z=3 or something

I used the functional form of Kamble et al. 2019 (1904.01110) which is pretty standard.

> And I'd maybe switch T[w] -> Fbar[w].

Agreed, I made the changes.